File: selection_menu_function.py
def selection_menu_function(action, SA):
    marked = [i for i, selected in enumerate(SA['Table']['Selected']) if selected]
    nonmarked = [i for i, selected in enumerate(SA['Table']['Selected']) if not selected]

    if action == 'Select marked shots':
        for i in marked:
            SA['Table']['Selected'][i] = True

    elif action == 'Unselect marked shots':
        for i in marked:
            SA['Table']['Selected'][i] = False

    elif action == 'Invert Selection':
        if marked:
            for i in marked:
                SA['Table']['Selected'][i] = False
            for i in nonmarked:
                SA['Table']['Selected'][i] = True
        else:
            print('Nothing marked, nothing to invert.')

    elif action == 'Unselect all shots':
        SA['Table']['Selected'] = [False] * len(SA['Table']['Selected'])

    elif action == 'Select all shots':
        SA['Table']['Selected'] = [True] * len(SA['Table']['Selected'])

    elif action == 'Delete selected shots':
        if marked:
            SA['Shot'] = [shot for i, shot in enumerate(SA['Shot']) if i not in marked]
            SA['Table'] = SA['Table'].drop(index=marked).reset_index(drop=True)
            print('Delete completed.')
        else:
            print('Nothing marked, nothing to delete.')

    elif action == 'Hide selected shots':
        if marked:
            print('Hide completed.')
        else:
            print('Nothing marked, nothing to hide.')

    # Update GUI
    update_shot_list(SA)

    # Plot selected things
    player_function('plotcurrent', SA)

File: selection_menu_function.m
function selection_menu_function(~, eventdata)

global SA

% uimenu(msel,'Text','Import from XLS','MenuSelectedFcn',@selection_menu_function);
% uimenu(msel,'Text','Select marked shots','MenuSelectedFcn',@selection_menu_function);
% uimenu(msel,'Text','Unselect marked shots','MenuSelectedFcn',@selection_menu_function);
% uimenu(msel,'Text','Delete selected shots','MenuSelectedFcn',@selection_menu_function);
% uimenu(msel,'Text','Select all shots','MenuSelectedFcn',@selection_menu_function);
% uimenu(msel,'Text','Unselect all shots','MenuSelectedFcn',@selection_menu_function);

h = findobj('Tag', 'uitable_shotlist');

marked = find(cell2mat(h.Data(:,1)));
nonmarked = find(cell2mat(h.Data(:,1))== false);

% Tableidcoli = find(SA.Table.Properties.VariableNames == 'ShotID');

col_si = strcmp('ShotID', h.ColumnName);
col_mi = strcmp('Mirrored', h.ColumnName);

for i = 1:length(marked)
    ShotID = cell2mat(h.Data(marked(i), col_si));
    mirrored = cell2mat(h.Data(marked(i), col_mi));
    
    [~, si, ~] = intersect(cell2mat(SA.Table.ShotID)+cell2mat(SA.Table.Mirrored)/10, ShotID+mirrored/10);
    
    markedDB(i) = si;
end

switch eventdata.Source.Text
    case 'Select marked shots'
        
        SA.Table.Selected(SA.Current_ti) = {true};
        
    case 'Unselect marked shots'
        
        SA.Table.Selected(SA.Current_ti) = {false};
        
    case 'Invert Selection'
        if ~isempty(marked)
            SA.Table.Selected(marked) = {false};
            SA.Table.Selected(nonmarked) = {true};
        else
            disp('Nothing marked, nothing to invert.')
        end
        
    case 'Unselect all shots'
        SA.Table.Selected = num2cell(false(size(SA.Table,1),1));
        
    case 'Select all shots'
        SA.Table.Selected = num2cell(true(size(SA.Table,1),1));
        
    case 'Delete selected shots'
        
        if ~isempty(marked) & ~isempty(markedDB)
            
            % delete rows which have a tick
            SA.Shot(markedDB) = [];
            SA.Table(markedDB,:) = [];
            SA.ShotIDsVisible(marked) = [];
            if length(SA.Shot) < SA.Current_si
                SA.Current_si = 1;
            end
            disp('Delete completed.')
        else
            disp('Nothing marked, nothing to hide.')
        end
        
    case 'Hide selected shots'
        
        if ~isempty(marked)
            % hide rows which have a tick
            SA.ShotIDsVisible(marked) = [];
            disp('hide completed.')
        else
            disp('Nothing marked, nothing to hide.')
        end
        
end

% update GUI
update_ShotList

% Plot selected things
PlayerFunction([],[])